components/cli/pkg/commands/image/test.go

```diff
@@ -221,221 +222,160 @@ func startTestCellInstance(imageDir string, instanceName string, runningNode *de
 	}
 
 	// Get Ballerina Installation if exists
-	moduleMgr := &util.BLangManager{}
-	exePath, err := moduleMgr.GetExecutablePath()
+	exePath, err := cli.BalExecutor().ExecutablePath()
 	if err != nil {
 		return fmt.Errorf("failed to get executable path, %v", err)
 	}
-	cmd := &exec.Cmd{}
 	verboseMode := strconv.FormatBool(verbose)
 
 	telepresenceYamlPath := filepath.Join(imageDir, "telepresence.yaml")
-	if !debug {
-		// Construct test artifact names
-		balProjectName := constants.TARGET_DIR_NAME
-		balTomlPath := filepath.Join(imageDir, constants.ZipBallerinaSource, constants.BallerinaToml)
-		testsPath := filepath.Join(imageDir, constants.ZipTests)
-		testsRoot := filepath.Join(currentDir, balProjectName)
-		var balModule string
-
-		if instanceName != "" {
-			balModule = filepath.Join(testsRoot, constants.ZipBallerinaSource, instanceName)
-		} else {
-			balModule = filepath.Join(testsRoot, constants.ZipBallerinaSource, constants.TempTestModule)
-		}
-		isTestDirExists, _ := util.FileExists(testsPath)
-		var isBallerinaProject bool
+	if debug && exePath == "" {
+		return fmt.Errorf("Ballerina should be installed to debug tests.")
+	}
+	balProjectName := constants.TARGET_DIR_NAME
+	// Construct test artifact names
+	if debug {
+		balProjectName = filepath.Base(projLocation)
+	}
 
-		//TODO: check if bal file contains @test
-		if !isTestDirExists {
-			return fmt.Errorf("no tests found in the cell image %v", imageTag)
-		}
+	balTomlPath := filepath.Join(imageDir, constants.ZipBallerinaSource, constants.BallerinaToml)
+	testsPath := filepath.Join(imageDir, constants.ZipTests)
+	testsRoot := filepath.Join(currentDir, balProjectName)
```

Comment:
Get currentDir from cli interface function "CurrentDir" so that it can be used for unit tests.
Otherwise CurrentDir is hard coded to os.GetWd(), dir in which the unit tests are run might get deleted.

Reply:
Fixed